ui/src/app/job-list/job-list.component.ts

```diff
@@ -33,7 +33,8 @@ export class JobListComponent implements OnInit {
   // whenever we change query parameters, via a subscription.
   jobs = new BehaviorSubject<JobListView>({
     results: [],
-    exhaustive: false
+    exhaustive: false,
+    needsRefresh: false
```

Comment:
So - problem here I think - `reloadJobs()` works by simply discarding the old job stream and creating a new one. The stream in the cache would be unmodified and we would not readd this to the cache. So the cached stream is just the first job stream for the given route.